spec/models/spree/order_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+require 'spree/order_decorator.rb'
+
+describe Spree::Order do
+  let(:order) { OrderWalkthrough.up_to(:payment) }
+  let(:stripe_apple_pay_payment_method) do
+    Spree::Gateway::StripeApplePayGateway.create!(
+      name: 'ApplePay',
+      preferred_domain_verification_certificate: 'dvc_test_jJG9kgDS75aADOsdfD1fDxQ'
+    )
+  end
+  let(:stripe_credit_card_payment_method) do
+    Spree::Gateway::StripeGateway.create!(
+      name: 'CreditCard',
+      preferred_secret_key: 'sk_test_aFQiNCcaOlXgDIwJHDdWlGN',
+      preferred_publishable_key: 'pk_test_ZItEjGEZxRiCuzGZHdsbOaj'
+    )
+  end
+
+  context 'when an order was paid with Apple Pay' do
+    before do
+      order.payments.create(payment_method_id: stripe_apple_pay_payment_method.id,
+                            amount: 29.99,
+                            state: 'completed')
+    end
+
+    it '#paid_with_apple_pay? returns true' do
```

Comment:
You should structure your tests differently , e.g 
```
describe '#paid_with_apple_pay` do
  context 'when an order was paid with Apply Pay' do
  ....
  end
end
```
http://www.betterspecs.org
https://github.com/reachlocal/rspec-style-guide

Reply:
fixing